src/decompress.js

```diff
@@ -0,0 +1,41 @@
+const zlib = require('zlib');
+
+const inflate = require('./inflate');
+
+function decompress(response, useBrotli) {
+    const compression = response.headers['content-encoding'] || 'identity';
+    const nodeVersion = Number(process.version.match(/^v(\d+\.\d+)/)[1]);
+
+    let decompressor;
+
+    switch (compression) {
+    case 'br':
+        if (!useBrotli) {
+            return response;
+        }
+
+        if (nodeVersion >= 12) {
+                decompressor = require('zlib').createBrotliDecompress(); // eslint-disable-line
+        } else {
+            try {
+                    decompressor = require('iltorb').decompressStream(); // eslint-disable-line
+            } catch (e) {
+                throw new Error('You must have iltorb peer dependency installed to use brotli decompression or use NodeJs v12+');
```

Comment:
But what if the error is thrown from `decompressStream()`? I'd add the try-catch just around `require()`

Reply:
I am not sure if it makes any difference. The result is still the same. `iltorb` is either not installed at all or it does not work, which has the same impact for the user -> reinstall/install the package.

Reply:
Well, if there is an invalid content in the stream, you'll probably throw the error "You must have iltorb peer dependency installed", which is actually not the correct description of the situation

Reply:
It does not decompress anything at this point. It just spawns the decompressor which is used. I think that it should throw error only once the stream is read. In case of default implementation, it should throw `'Could not convert stream to string'` error. If the user wants a stream to be returned. It throws once he is reading the stream by some function.